src/lxml/html/tests/test_basic.py

```diff
@@ -1,11 +1,51 @@
+import sys
 import unittest
 from lxml.tests.common_imports import make_doctest, doctest
-import lxml.html
+from lxml import html
+
+class TestBasicFeatures(unittest.TestCase):
+    def test_various_mixins(self):
+        base_url = "http://example.org"
+        doc = html.fromstring("""
+        <root>
+            <!-- comment -->
+            <?pi contents ?>
+            &entity;
+            <el/>
+        </root>
+        """, base_url=base_url)
+        self.assertEqual(doc.getroottree().docinfo.URL, base_url)
+        self.assertEqual(len(doc), 3)
+        self.assertIsInstance(doc[0], html.HtmlComment)
+        self.assertIsInstance(doc[1], html.HtmlProcessingInstruction)
+        self.assertIsInstance(doc[2], html.HtmlElement)
+        for child in doc:
+            # base_url makes sense on all nodes (kinda) whereas `classes` or
+            # `get_rel_links` not really
+            self.assertEqual(child.base_url, base_url)
+
+    def test_set_extension(self):
```

Comment:
Not sure what you mean with "extension" here.
```suggestion
    def test_set_empty_attribute(self):
```

Reply:
> Not sure what you mean with "extension" here.

I was thinking about the feature being added by / through the HtmlMixin (compared to the regular Element which is what I usually interact with when using lxml) but it's true that since the tests are already in the html context it doesn't make much sense.

Reply:
src/lxml/html/__init__.py

Reply:
> src/lxml/html/__init__.py

https://github.com/lxml/lxml/blob/master/.travis.yml

Reply:
> Not sure what you mean with "extension" here.
> ``https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975
>     def test_set_empty_attribute(self):
> ```

Reply:
> Not sure what you mean with "extension" here.
> ```suggestion
>     def test_set_empty_attribute(self):
> ```

Reply:
> > src/lxml/html/__init__.py
> https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975
>https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975 https://github.com/lxml/lxml/blob/master/.travis.yml

Reply:
> Not sure what you mean with "extension" here.
> ```suggestion
>     def test_set_empty_attribute(self):
> ```

https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975

Reply:
> > src/lxml/html/__init__.py
> https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975
> https://github.com/lxml/lxml/blob/master/.travis.yml

Reply:
> Not sure what you mean with "extension" here.
> ```suggestion
>     def test_set_empty_attribute(self):
> ```

8132c755adad4a75ba855d985dd257493bccc7fd

Reply:
> > src/lxml/html/__init__.py
> 8132c755adad4a75ba855d985dd257493bccc7fd
> https://github.com/lxml/lxml/blob/master/.travis.yml

Reply:
> > Not sure what you mean with "extension" here.
> 
> I was thinking about the feature being added by / through the HtmlMixin (compared to the regular Element which is what I usually interact with when using lxml) but it's true that since the tests are already in the html context it doesn't make much sense.

Reply:
> Not sure what you mean with "extension" here.
> ```suggestion
>     def test_set_empty_attribute(self):
> ```

Reply:
> > Not sure what you mean with "extension" here.
> > ``https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975
> >     def test_set_empty_attribute(self):
> > ```
> 
> 

https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975

Reply:
> Not sure what you mean with "extension" here.
> ```suggestion
>     def test_set_empty_attribute(self):
> ```
https://github.com/lxml/lxml/pull/340#pullrequestreview-904505975